reactor-core/src/main/java/reactor/core/publisher/FluxTakeUntil.java

```
@@ -88,13 +93,16 @@ public void onNext(T t) {
 
 			if (b) {
 				s.cancel();
-
-				onComplete();
+				took = true;
+				actual.onComplete();
 			}
 		}
 
 		@Override
 		public void onError(Throwable t) {
+			if (took) {
+				return;
```

why would this case silently drop the error ?